### DISCUSSION TITLE
Performance Issue with ArgoCD UI

### DISCUSSION THREAD

**Discussion by naveengoyal1811** [argoproj/argo-cd#10238] [General]
I am facing issue with ArgoCD UI (its running very slow) when deploying a helm chart for multiple clusters (using appset), Could you please help on this issue.. also could see that application-controller pod is consuming very high CPU,  are there any variables which can improve this UI performance, any info would be great help..  thank you !!

**Comment by crenshaw-dev** [comment]
When you say the UI is slow... it's slow when viewing what? Sounds like you're viewing a Helm Application which contains one or more ApplicationSet resources. Does it contain anything else?

**Comment by naveengoyal1811** [comment]
It contains applicationset resources only to deploy multiple helm charts on multiple clusters. So when I click on refresh or sync buttons, it is taking high time there or sometimes keep loading..

**Reply by crenshaw-dev** [reply]
Okay I think I've experienced the same thing. The UI kinda blinks a bunch, and sometimes click events seem to be ignored. I'll hopefully have time to look into it soon.

**Comment by naveengoyal1811** [comment]
Thank you.. 
please let me know if I can provide any more information from my end.

**Reply by naveengoyal1811** [reply]
Hi @crenshaw-dev ,
Just curious to know if you got some time to analyse or fix the issue as it is blocking us to proceed with our solution. Thanks!

**Comment by jeongsm2** [comment]
Same issue here. I hope this problem to solve soon. 

**More information**
> argo-cd chart version: `argo-cd-4.10.6`
> argo-cd app version: `v2.4.9`

Additionally, I recognized adding some monitoring related items for orphan resources to AppProject, the time taking up in the controller increased. But running without this options, it increases the waiting time for a response after clicking the Sync button in the UI very long.

**Comment by naveengoyal1811** [comment]
@crenshaw-dev,
Did you find out anything?
Could you please provide any update. Thanks.

**Comment by mubarak-j** [comment]
Have tried using gzip compression? It can be enabled via [argocd-server args](https://github.com/argoproj/argo-helm/blob/7992b5d04b9ea0d13a3ae7799e042cf9056eb6cd/charts/argo-cd/values.yaml#L1303) (if using community helm-chart)

```yaml
server:
  extraArgs:
  - --enable-gzip
 ```

See this issue for more details: https://github.com/argoproj/argo-cd/issues/4226

**Reply by Aracki** [reply]
even with this, it's being slow with ~1500-2000 applications 😢
